packages/app/src/messages/selectors/get-thread-draft-messages.selector.test.ts

```diff
@@ -0,0 +1,155 @@
+  content: "Outbox message",
+  threadId: "1",
+  phoneNumber: "+48 755 853 216",
+  messageType: MessageType.OUTBOX,
+}
+
+const messagesState: MessagesState = {
+  threadMap: {
+    [thread.id]: thread,
+  },
+  messageMap: {
+    [firstDraftMessage.id]: firstDraftMessage,
+    [outboxMessage.id]: outboxMessage,
+  },
+  messageIdsInThreadMap: {
+    [thread.id]: [
+      firstDraftMessage.id,
+      secondDraftMessage.id,
+      outboxMessage.id,
+    ],
+  },
+  searchValue: "",
+  visibilityFilter: VisibilityFilter.All,
+  threadsState: ResultState.Empty,
+  messagesStateMap: {},
+  error: null,
+  loaded: false,
+  loading: false,
+  currentlyDeletingMessageId: null,
+}
+
+describe("When thread hasn't any message", () => {
+  test("returns `undefined`", () => {
+    const state = {
+      messages: {
+        ...messagesState,
+        messageMap: {},
+        messageIdsInThreadMap: {
+          ...messagesState.messageIdsInThreadMap,
+          [thread.id]: [],
+        },
+      },
+    } as ReduxRootState
+    expect(getThreadDraftMessagesSelector(thread.id)(state)).toBeUndefined()
+  })
+})
+
+describe("When thread have only `OUTBOX` messages", () => {
+  test("returns `undefined`", () => {
+    const state = {
+      messages: {
+        ...messagesState,
+        messageMap: {
+          [outboxMessage.id]: outboxMessage,
+        },
+        messageIdsInThreadMap: {
+          ...messagesState.messageIdsInThreadMap,
+          [thread.id]: [outboxMessage.id],
+        },
+      },
+    } as ReduxRootState
+    expect(getThreadDraftMessagesSelector(thread.id)(state)).toBeUndefined()
+  })
+})
+
+describe("When thread have `DRAFT` messages", () => {
+  test("returns draft message if one exists", () => {
+    const state = {
+      messages: {
+        ...messagesState,
+        messageMap: {
+          [firstDraftMessage.id]: firstDraftMessage,
+          [outboxMessage.id]: outboxMessage,
+        },
+        messageIdsInThreadMap: {
+          ...messagesState.messageIdsInThreadMap,
+          [thread.id]: [outboxMessage.id, firstDraftMessage.id],
+        },
+      },
+    } as ReduxRootState
+    expect(getThreadDraftMessagesSelector(thread.id)(state)).toEqual(
+      firstDraftMessage
+    )
+  })
+
+  test("returns last draft message if many exists", () => {
+    const state = {
+      messages: {
+        ...messagesState,
+        messageMap: {
+          [firstDraftMessage.id]: firstDraftMessage,
+          [secondDraftMessage.id]: firstDraftMessage,
+          [outboxMessage.id]: outboxMessage,
+        },
+        messageIdsInThreadMap: {
+          ...messagesState.messageIdsInThreadMap,
+          [thread.id]: [
+            outboxMessage.id,
+            firstDraftMessage.id,
+            firstDraftMessage.id,
+          ],
+        },
+      },
+    } as ReduxRootState
+    expect(getThreadDraftMessagesSelector(thread.id)(state)).toEqual(
+      firstDraftMessage
+    )
+  })
```

Comment:
Is it ok that there is such a mapping 
```
          [firstDraftMessage.id]: firstDraftMessage,
          [secondDraftMessage.id]: firstDraftMessage,
```
and
```
            outboxMessage.id,
            firstDraftMessage.id,
            firstDraftMessage.id,
```
? Shouldn't it be

```
          [firstDraftMessage.id]: firstDraftMessage,
          [secondDraftMessage.id]: secondDraftMessage,
```

and 

```
            outboxMessage.id,
            firstDraftMessage.id,
            secondDraftMessage.id,
```
?

Reply:
@wojcik92michal I don't understand what exactly you mean here?

Reply:
https://github.com/mudita/mudita-center/pull/835/commits/31b4aae7dbdc63e21c41fb1e179a2ea58b9d9d98